src/mpi/comm/commutil.c

```diff
@@ -287,6 +287,9 @@ int MPIR_Comm_create(MPIR_Comm ** newcomm_ptr)
     mpi_errno = MPII_Comm_init(newptr);
     MPIR_ERR_CHECK(mpi_errno);
 
+    mpi_errno = MPID_Comm_create(newptr);
+    MPIR_ERR_CHECK(mpi_errno);
+
```

Comment:
The reason is flawed. 

Why not setting up your states in comm_commit_pre_hook ?

Reply:
If the state is set in comm_commit_pre_hook (which was already the case), we have the following problem:
- `comm_idup_with_info` reaches `update_nic_preferences` before calling `comm_commit_pre_hook`.  - 
- `comm_commit_pre_hook` is supposed to sets `pref_nic` to NULL.  Before that, this pointer has an invalid value.
- As a result, `update_nic_preferences` does not see `NULL` in `pref_nic` and assumes that it is already allocated. Then it tries to access values from `pref_nic`, and we get a segfault. 

This solution was added as a way to mitigate this problem, i.e., track if pref_nic has a valid value. Do you have any idea for an alternative solution?

Reply:
Added the change that we discussed - to distinguish whether the hints were set from comm_create or set_info